common/presenters/events-to-timeline/event-details-helpers.js

```diff
@@ -0,0 +1,84 @@
+const pluralize = require('pluralize')
+
+const { event: eventModel } = require('../../../common/lib/api-client/models')
+const i18n = require('../../../config/i18n')
+
+const eventRelationships = Object.keys(eventModel.fields).filter(field => {
+  return typeof eventModel.fields[field] === 'object'
+})
+
+// add resources that belong to move and are common to all events
+// NB. move.supplier will exist even when event.supplier does not
+const getMoveDetails = move => ({
+  move,
+  person: move.profile?.person,
+})
+
+// if an event has a supplier, the supplier was the agent
+// if not, check the locales for a default agent
+// finally, use PMU as default agent
+const getEventAgency = event => {
+  let agency = event.supplier ? 'supplier' : undefined
+
+  if (!agency) {
+    const defaultAgencyKey = `events::${event.event_type}.defaultAgency`
+    agency = i18n.exists(defaultAgencyKey) ? i18n.t(defaultAgencyKey) : 'pmu'
+  }
+
+  return { agency }
+}
+
+// add context (if any) to event
+const getEventContext = event => {
+  const contextKeyDefinition = `events::${event.event_type}.contextKey`
+
+  const contextKey = i18n.exists(contextKeyDefinition)
+    ? i18n.t(contextKeyDefinition)
+    : undefined
+
+  const context = event.details?.[contextKey]
+  return { context }
+}
+
+// add event relationships
+const getEventRelationships = event => {
+  return eventRelationships.reduce((acc, rel) => {
+    if (event[rel]) {
+      acc[rel] = event[rel]
+    }
+
+    return acc
+  }, {})
+}
+
+// add eventable resources to details object
+// eg event.eventable.type === 'journeys'
+// => event.details.journey
+const getEventableDetails = event => {
+  const details = {}
+
+  if (event.eventable && event.eventable.type) {
+    details[pluralize.singular(event.eventable.type)] = event.eventable
+  }
+
+  return details
+}
+
+// add standard common properties of the event itself
+const getEventProperties = event => {
+  // eslint-disable-next-line camelcase
+  const { notes, occurred_at } = event
```

Comment:
Does this `occurred_at` property get passed into the timeline component?

Reply:
Not as is, no.

It gets used

a) to set the timestamp on the event item passed in to the timeline component

`const timestamp = event.occurred_at`
https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/d159d58e4d9537ff31c5246524d795dbaf89dd1a/common/presenters/events-to-timeline-component.js#L50

b) part of the vars passed into strings for interpolation by i18next

eg. 
`"description": "{{person.fullname}} was offered $t(events::PerPrisonerWelfare.select_subtype) at {{occurred_at, formatTime}}...`
https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/d159d58e4d9537ff31c5246524d795dbaf89dd1a/locales/en/events.json#L336